HotelBase/HotelBaseClass.cs

```diff
@@ -54,5 +58,54 @@ internal bool BookRoomInExternalService(DateTime date)
         {
             return _bookingService.Book(date);
         }
+
+        public ReservationResult Reserve(DateTime date, double price, int creditCardNumber, string email)
+        {
+            bool paymentMade = false;
+            bool roomBooked = false;
+            bool emailSent = false;
+            bool success = true;
+            bool priceValid = false;
+            string reservationNumber = null;
+
+            while (Operations.Count != 0)
+            {
+                HotelOperation currentOperation = Operations.OrderBy(o => o.Order).First();
+                bool stepSuccess = true;
+
+                switch (currentOperation.Operation)
```

Comment:
To consider:

Did you thing about use case where operation is not one of initial defined? For instance someone add a new operation, but forgot to extent this method? What should happen then. Now this operation will not be executed, which maybe cool, I don't know, I just wan to make sure that it was your intention

Reply:
Partially it was my intention (since ReservationResult would need to be extended as well if we'd like to know whether this new operation was successful or not), but on the other side I'm not even sure how it could be implemented (passing the "custom" method). I'd think about it